cypress/integration/site-search/index.spec.ts

```diff
@@ -0,0 +1,39 @@
+describe('Docs website search', () => {
+  let searchTerm = '';
+  before(() => {
+    // Start an XHR server with the built-in cypress utilities.
+    cy.server();
+    cy.route('**/**/page-data.json').as('pageDataResponse');
+    cy.route('**/**/app-data.json').as('appDataResponse');
+    cy.visit('/');
+    // Wait for the JS assets to load since they are necessary for the search functionality.
+    cy.wait(['@pageDataResponse', '@appDataResponse']);
```

Comment:
Running locally, removing the wait here has no impact on the test run. Inspecting the network requests I can't see any `app-data.json` files, and `page-data.json` is only page content unrelated to search. 

I think it's super safe to remove them.

Reply:
@SiTaggart Without these wait statements, we get an error from cypress that the input cannot be found because it has been temporarily removed from the dom. I didn't dig too much into it, but AFAIK it's related to the state/status of these requests, since if you block these requests, the components that require JS for interaction do not work.
Unfortunately, this is definitely tightly coupled to a gatsby internal, and I know we should avoid that.

I noticed on the navigation tests, we have an arbitrary wait for some milieseconds, which is a patch for this kind of issue, but is [regarded as an antipattern by the cypress folks](https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting).